scripts/mkcloud

```diff
@@ -1340,57 +1334,13 @@ function sanity_checks()
-            # currently only drbd with 2 nodes supported
-            nodenumberdatacluster=2
-            local maxnodes=$(( nodenumber - nodenumbercompute - nodenumberdatacluster ))
-
-            # calculating cluster size
-            # rule1: clustersize % 2 == 1 && clustersize >= 3
-            # rule2: 2 * clustersize + offset = maximum_number_of_nodes
-            # first calculating the offset (may be 0, 1, 2 or 3)
-            nodenumbercomputeoffset=$(( ( maxnodes + 2 ) % 4 ))
-            # new maximum number (double of clustersize)
-            local mn=$(( maxnodes - nodenumbercomputeoffset ))
-            # cluster size found
-            nodenumbernetworkcluster=$(( mn / 2 ))
-            nodenumberservicescluster=$(( mn / 2 ))
-        fi
-
-        I="    "
-        echo "== HA node number overview =="
-        echo  "configured nodes (min. nodes) : use case"
-        echo                "$I$nodenumber (9) : nodenumber"
-        echo         "$I$nodenumbercompute (1) : nodenumber compute"
-        echo   "$I$nodenumbercomputeoffset (0) : unused nodes added as compute nodes"
-        echo "Resulting HA node numbers:"
-        echo     "$I$nodenumberdatacluster (2) : nodenumber data cluster"
-        echo  "$I$nodenumbernetworkcluster (3) : nodenumber network cluster"
-        echo "$I$nodenumberservicescluster (3) : nodenumber services cluster"
-
-        # sanity check
-        if  [[ $(( $nodenumbercompute + $nodenumbercomputeoffset )) -lt 1  ]] || \
-            [[ $(( $nodenumberservicescluster )) -lt 3 ]] || \
-            [[ $(( $nodenumbernetworkcluster  )) -lt 3 ]] || \
-            [[ $nodenumberdatacluster -lt $nodenumberdataclusterdefault ]] ; then
-            echo
-            echo "Error: too few nodes for a HA cloud setup"
-            echo "Please redefine the number of nodes:"
-            echo " export nodenumber=<x>         and/or"
-            echo " export nodenumbercompute=<y>"
-            echo
-            echo "Note:"
-            echo " For a quorum a cluster needs to have an odd number of nodes, at least 3."
-            echo " For drbd cluster only 2 nodes are supported currently."
-            echo " So your nodenumber shoud be:"
-            echo "     compute_node_number + 2 (drbd cluster) + 2 * odd_number (cluster)"
-            complain 88 "Please raise the number of nodes via e.g. export nodenumber=10"
-        fi
-
-        nodenumbercompute=$(( $nodenumbercompute + $nodenumbercomputeoffset ))
-        echo
+    # checking clusterconfig
+    if [ -n "$hacloud" -a -z "$clusterconfig" ] ; then
+        echo "Examples for clusterconfig:"
+        echo '3 clusters: clusterconfig="data=2:network=3:services=3"'
+        echo '2 clusters: clusterconfig="services+data=2:network=3"'
+        echo '1 cluster:  clusterconfig="data+network+services=2"'
+        complain 72 "No cluster config provided for HA setup."
```

Comment:
We are limited to 127 exit codes. So at some point in time we will have duplication. So they are not meant to be unique. The idea behind these exit codes (with complain) is just to limit the search effort.
But I agree that we should take another number as long as we have some :)

FYI: There are also usages like "complain $? ...", they will always have an unpredictable exit code. So you should really compare the message from the log.